.github/workflows/main.yml

```diff
@@ -34,6 +34,24 @@ env:
 
 
 jobs:
+  pyright:
+    strategy:
+      matrix:
+        python:
+        - "3.8"
+        - "3.11"
+        - "3.12"
+        platform:
+        - ubuntu-latest
+        - macos-latest
+        - windows-latest
+    runs-on: ${{ matrix.platform }}
+    steps:
+      - uses: actions/checkout@v4
+      - uses: jakebailey/pyright-action@v2
+        with:
+          python-version: ${{ matrix.python }}
+
```

Comment:
I know this isn't how you usually do your tests. I'm using it as a placeholder. I need a hand ensuring mypy and pyright are run as part of your testing setup.

Reply:
Hi @Avasam , does simply adding `pytest-mypy` to the `testing` `extras_require` config (in `setup.cfg` I suspect) does not work for now?

Reply:
It's already there https://github.com/pypa/setuptools/blob/main/setup.cfg#L50 , so if it was used, it should already have been failing.
Unless the CI is running on PyPy ?

Reply:
I figured it out, it was indeed never run, you need to explicitly specify the `--mypy` flag to pytest. Also this installs and runs fine on PyPy now (dropping Python 3.7 support and AST from Mypy probably helped). Other than https://github.com/abravalheri/validate-pyproject/pull/159

Reply:
On the note of pytest integration:
I just tried out `pytype-pyright` but it didn't fail where `mypy`, `pytest-mypy`, `jakebailey/pyright-action` and `pyright` did, even after setting `--pyright-dir .` on the pytest command line. So I'm not sure I can trust it for this use (it was also developed for internal use in a different project)